homeassistant/components/kostal_plenticore/switch.py

```diff
@@ -19,39 +20,35 @@
 _LOGGER = logging.getLogger(__name__)
 
 
-class SwitchData(NamedTuple):
-    """Representation of a SelectData tuple."""
+@dataclass
+class PlenticoreRequiredKeysMixin:
+    """A class that describes required properties for plenticore switch entities."""
 
     module_id: str
-    data_id: str
-    name: str
     is_on: str
     on_value: str
     on_label: str
     off_value: str
     off_label: str
 
 
-# Defines all entities for switches.
-#
-# Each entry is defined with a tuple of these values:
-#  - module id (str)
-#  - process data id (str)
-#  - entity name suffix (str)
-#  - on Value (str)
-#  - on Label (str)
-#  - off Value (str)
-#  - off Label (str)
+@dataclass
+class PlenticoreSwitchEntityDescription(
+    SwitchEntityDescription, PlenticoreRequiredKeysMixin
+):
```

Comment:
In this PR, I am only converting the `NamedTuple` to `SwitchEntityDescription`.
The old `data_id` is mapped to `key`, and `name` is part of the standard `EntityDescription`.

Passing the `EntityDescription` to `PlenticoreDataSwitch` will be done in a follow-up PR.